e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/deal-cancellation-enabled/dashboard-facilities-filter-by-facility-stage.spec.js

```diff
@@ -0,0 +1,200 @@
+const { FACILITY_STATUS, FACILITY_STAGE } = require('@ukef/dtfs2-common');
+const relative = require('../../../../relativeURL');
+const MOCK_USERS = require('../../../../../../../e2e-fixtures');
+const CONSTANTS = require('../../../../../fixtures/constants');
+const { dashboardFacilities } = require('../../../../pages');
+const { dashboardFilters } = require('../../../../partials');
+const {
+  BSS_DEAL_AIN,
+  BSS_FACILITY_BOND_ISSUED,
+  BSS_FACILITY_BOND_UNISSUED,
+  BSS_FACILITY_BOND_ISSUED_RISK_EXPIRED,
+  BSS_FACILITY_BOND_UNISSUED_RISK_EXPIRED,
+} = require('../../fixtures');
+const {
+  submitRedirectsToDashboard,
+  shouldRenderCheckedCheckbox,
+  shouldRenderAppliedFilterInPanelSelectedFilters,
+  shouldRenderAppliedFilterInMainContainerSelectedFilters,
+} = require('../_actions-and-assertions');
+
+const { BANK1_MAKER1, ADMIN } = MOCK_USERS;
+
+const filters = dashboardFilters;
+
+context('Dashboard Facilities filters - filter by facility stage', () => {
+  const ALL_FACILITIES = [];
+
+  before(() => {
+    cy.deleteGefApplications(ADMIN);
+    cy.deleteDeals(ADMIN);
+
+    cy.insertOneDeal(BSS_DEAL_AIN, BANK1_MAKER1).then((deal) => {
+      const dealId = deal._id;
+
+      const facilities = [BSS_FACILITY_BOND_ISSUED, BSS_FACILITY_BOND_UNISSUED, BSS_FACILITY_BOND_ISSUED_RISK_EXPIRED, BSS_FACILITY_BOND_UNISSUED_RISK_EXPIRED];
+
+      cy.createFacilities(dealId, facilities, BANK1_MAKER1).then((insertedFacilities) => {
+        insertedFacilities.forEach((facility) => {
+          ALL_FACILITIES.push(facility);
+        });
+      });
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+    dashboardFacilities.visit();
+
+    // toggle to show filters (hidden by default)
+    filters.showHideButton().click();
+  });
+
+  describe(CONSTANTS.FACILITY.FACILITY_STAGE.ISSUED, () => {
```

Comment:
Facility stage is really weird - the value stored in the database `facilityStage` has historically only been on BSS/EWCS deals taking the values `conditional` or `unconditional`. I've added to this field `Risk expired` & this is also on GEF facilities too.

The `FACILITY_STAGE` constant is representing whats stored in the database so doesnt have `Issued` or `unissued` - these are frontend only terms, decided by the boolean `hasBeenIssued`. This is a bit annoying & not very extensible but it would be a huge refactor to update that